falco-plugin/main.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"time"
+
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/extractor"
+	"github.com/falcosecurity/plugin-sdk-go/pkg/sdk/plugins/source"
+	"google.golang.org/grpc"
+
+	pb "github.com/kinvolk/seccompagent/falco-plugin/api"
+)
+
+// SeccompAgentPlugin represents our plugin
+type SeccompAgentPlugin struct {
+	plugins.BasePlugin
+
+	SocketFile    string `json:"socketFile" jsonschema:"description=Socket File for receiving events from Seccomp Agent via gRPC (Default: /run/seccomp-agent-falco-plugin.sock)"`
+	FlushInterval uint64 `json:"flushInterval" jsonschema:"description=Flush Interval in ms (Default: 30)"`
+}
+
+// SeccompAgentInstance represents a opened stream based on our Plugin
+type SeccompAgentInstance struct {
+	source.BaseInstance
+	msgC <-chan SeccompAgentMessage
+	errC <-chan error
+	ctx  context.Context
+}
+
+// init function is used for referencing our plugin to the Falco plugin framework
+func init() {
+	p := &SeccompAgentPlugin{}
+	extractor.Register(p)
+	source.Register(p)
+}
+
+// Info displays information of the plugin to Falco plugin framework
+func (seccompAgentPlugin *SeccompAgentPlugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:                 5,
+		Name:               "seccompagent",
+		Description:        "Seccomp Agent Events",
+		Contact:            "github.com/kinvolk/seccompagent/",
+		Version:            "0.1.0",
+		RequiredAPIVersion: "0.3.0",
+		EventSource:        "seccompagent",
```

Comment:
Which `ID` should I use?

Reply:
Maybe we can switch to KSA, KSecAgent or something else? It can be Kinvolk Seccomp Agent, but the K can be also Kubernetes.

I think KSA might be too used (like Kubernetes Service Account, Kubernetes Security Announce), but this seems to ask for something shorter

Reply:
To sync with https://github.com/falcosecurity/plugins/pull/73